File: frontend/src/types/resource.ts
export interface Tag {
  id: number;
  name: string;
}

export interface Resource {
  id: number;
  title: string;
  description: string;
  type: string;
  url: string;
  created_at: string;
  tags: Tag[];
}

export interface PaginatedResponse<T> {
  data: T[];
  total: number;
  page: number;
  page_size: number;
}

File: backend/app/models/resource.py
from sqlalchemy import Column, Integer, String, Text, DateTime
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.db.base import Base


class Resource(Base):
    __tablename__ = "resources"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(255), nullable=False)
    description = Column(Text)
    type = Column(String(50), nullable=False)
    url = Column(String(500), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    tags = relationship("Tag", secondary="resource_tags", back_populates="resources")
